### PR TITLE
beast: migrate from homebrew-science

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
https://github.com/Homebrew/homebrew-science/pull/6341

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/beast.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  url "https://github.com/beast-dev/beast-mcmc/archive/v1.8.4.tar.gz"
+  sha256 "de8e7dd82eb9017b3028f3b06fd588e5ace57c2b7466ba2e585f9bd8381407af"
+  head "https://github.com/beast-dev/beast-mcmc.git"
+
+  depends_on :ant => :build
+  depends_on :java => "1.7+"
+
+  def install
+    system "ant", "linux"
+    prefix.install Dir["release/Linux/BEASTv*/*"]
```
Comment (ilovezfs) [inline Formula/beast.rb]: I'd prefer we put all of this in libexec and then symlink as needed elsewhere.

File: Formula/beast.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pkgshare.install prefix/"examples"
+
+    # Move installed JARs to libexec
+    mv lib, libexec
+
+    # Point wrapper scripts to libexec
```
Comment (ilovezfs) [inline Formula/beast.rb]: This probably won't be needed if you maintain the directory structure.

File: Formula/beast.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    Examples are installed in:
+      #{opt_pkgshare}/examples/
+    EOS
+  end
+
+  test do
+    cp (opt_pkgshare/"examples"/"clockModels"/"testUCRelaxedClockLogNormal.xml"), testpath
```
Comment (ilovezfs) [inline Formula/beast.rb]:
```
    cp pkgshare/"examples/clockModels/testUCRelaxedClockLogNormal.xml", testpath
```

File: Formula/beast.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    # Run fewer generations to speed up tests
+    inreplace "testUCRelaxedClockLogNormal.xml" do |s|
+      s['chainLength="10000000"'] = 'chainLength="500000"'
```
Comment (ilovezfs) [inline Formula/beast.rb]: gsub!

Comment (ilovezfs) [reply Formula/beast.rb]: but since it's a single inreplace, I don't think we need the block form at all.

File: Formula/beast.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    system "#{bin}/beast", "-beagle_off", "testUCRelaxedClockLogNormal.xml"
+
+    %w[ops log trees].each do |ext|
+      assert File.exist? "testUCRelaxedClockLogNormal." + ext
```
Comment (ilovezfs) [inline Formula/beast.rb]: assert_predicate

File: Formula/beast.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  depends_on :ant => :build
+  depends_on :java => "1.7+"
+
+  def install
+    system "ant", "linux"
+    prefix.install Dir["release/Linux/BEASTv*/*"]
+    pkgshare.install prefix/"examples"
+
+    # Move installed JARs to libexec
```
Comment (ilovezfs) [inline Formula/beast.rb]: moving the files twice seems weird. see comment above

File: Formula/beast.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    mv lib, libexec
+
+    # Point wrapper scripts to libexec
+    inreplace Dir[bin/"*"] do |s|
+      s["$BEAST/lib"] = "$BEAST/libexec"
+    end
+  end
+
+  def caveats; <<-EOS.undent
```
Comment (ilovezfs) [inline Formula/beast.rb]: don't need this caveat

**Comment by jonchang** [comment]
Ok, addressed all review comments. Homebrew doesn't like it when the `bin` folder is a symlink so I had to link each executable individually.

File: Formula/beast.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  head "https://github.com/beast-dev/beast-mcmc.git"
+
+  depends_on :ant => :build
+  depends_on :java => "1.7+"
+
+  def install
+    system "ant", "linux"
+    libexec.install Dir["release/Linux/BEASTv*/*"]
+    mkdir_p [pkgshare, bin]
+    ln_s libexec/"examples", pkgshare
```
Comment (ilovezfs) [inline Formula/beast.rb]: use .install_symlink

File: Formula/beast.rb [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 "de8e7dd82eb9017b3028f3b06fd588e5ace57c2b7466ba2e585f9bd8381407af"
+  head "https://github.com/beast-dev/beast-mcmc.git"
+
+  depends_on :ant => :build
+  depends_on :java => "1.7+"
+
+  def install
+    system "ant", "linux"
+    libexec.install Dir["release/Linux/BEASTv*/*"]
+    mkdir_p [pkgshare, bin]
```
Comment (ilovezfs) [inline Formula/beast.rb]: use mkpath

Comment (ilovezfs) [reply Formula/beast.rb]: actually it won't be needed if you use `.install_symlink`

File: Formula/beast.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    ln_s libexec/"examples", pkgshare
+    ln_s Dir[libexec/"bin/*"], bin
```
Comment (ilovezfs) [inline Formula/beast.rb]: use .install_symlink

File: Formula/beast.rb [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  test do
+    cp pkgshare/"examples"/"clockModels"/"testUCRelaxedClockLogNormal.xml", testpath
+
+    # Run fewer generations to speed up tests
+    inreplace "testUCRelaxedClockLogNormal.xml", 'chainLength="10000000"', 'chainLength="100000"'
+
+    system "#{bin}/beast", "-beagle_off", "testUCRelaxedClockLogNormal.xml"
+
+    %w[ops log trees].each do |ext|
+      assert_predicate testpath/("testUCRelaxedClockLogNormal." + ext), :exist?
```
Comment (ilovezfs) [inline Formula/beast.rb]: should have a custom message to print on failure

File: Formula/beast.rb [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    mkdir_p [pkgshare, bin]
+    ln_s libexec/"examples", pkgshare
+    ln_s Dir[libexec/"bin/*"], bin
+  end
+
+  test do
+    cp pkgshare/"examples"/"clockModels"/"testUCRelaxedClockLogNormal.xml", testpath
+
+    # Run fewer generations to speed up tests
+    inreplace "testUCRelaxedClockLogNormal.xml", 'chainLength="10000000"', 'chainLength="100000"'
```
Comment (ilovezfs) [inline Formula/beast.rb]:
```
    inreplace "testUCRelaxedClockLogNormal.xml", 'chainLength="10000000"',
                                                 'chainLength="100000"'
```
to keep line length < 80 columns

**Comment by jonchang** [comment]
Ok! `install_symlink` makes it much nicer.

**Comment by ilovezfs** [comment]
Thanks @jonchang! One down, five million to go.

**Pull request closed without merging**